### ISSUE TITLE
Crashes early in a few games

### ISSUE THREAD

**Issue by anamegoeshere23** [spiritualized1997/openFPGA-GBA#42]
-Sword of Mana - Graphical glitches followed by freezing and intermittent audio screeching during the cutscene after starting game and choosing character.  
-Super Puzzle Fighter 2 - crashes during first round
-Castlevania Aria of Sorrow - crashes early
-Golden Sun - crashes while running away

Spiritualized1.2.0, Pocket Firmware 1.1, bios present, sandisk Extreme, Black pocket received Augustish
Edit: Someone else on the internet tried Sword of Mana and said it worked, but I checksum'd my bios and game file - they checked out.  Interestingly they work for me with a flash cart but not through spiritualized.  Other cores seem to work as expected.

**Comment by execution101** [comment]
[DoomIISpiritualized.GBA_20230904_144809.txt](https://github.com/spiritualized1997/openFPGA-GBA/files/12517037/DoomIISpiritualized.GBA_20230904_144809.txt)
[CastlevaniaAriaOfSorrowSpiritualized.GBA_20230904_144628.txt](https://github.com/spiritualized1997/openFPGA-GBA/files/12517038/CastlevaniaAriaOfSorrowSpiritualized.GBA_20230904_144628.txt)
First I want to say thank you for making this core. I hope in time you can make some tweaks to it.
I would also like to add to this list
-Mother 3 English translation 1.2 and 1.3 crash before even getting through the first cut scene loud audio noises start when it crashes and the screen freezes.   
-I also tried Castlevania Aria of Sorrow crashes very quickly when trying to start the game I've attached a log file from the pocket regarding
-Doom 2 also crashes can't get past the main menu of the game
I attached some logs regarding

**Comment by azurehours** [comment]
Having this problem as well with multiple games. Metroid Zero Mission, Golden Sun: The Lost Age, Rebel Star, etc.
Seems to crash on more games than not. Other cores I've tried have had no issue.
Have a GITD edition.

**Comment by DunhamGitHub** [comment]
(Maybe I fail to see after reading all these posts here 3 times) ...do you guys use physical cartridges or the Cores?

* because if it is cores, it's probably related to ill-formatted / cheap SD-cards. Also in this case you could have (should have) dropped the CRC32 of some ROMs do others (like me) can check them (because there are various different versions / dumps of games)...

**Comment by azurehours** [comment]
The core for me. I've used a SanDisk Extreme Pro 256GB (UHS-I, A2) for my SD card. Reformatted in ExFAT, did a clean core install, same deal
CRC32s for the mentioned ROMs, for anyone interested

Metroid Zero Mission:
5c61a844

Rebelstar: Tactical Command:
fb6c590f

Golden Sun: The Lost Age:
606a1c4d

**Comment by DunhamGitHub** [comment]
> The core for me. I've used a SanDisk Extreme Pro 256GB (UHS-I, A2) for my SD card. Reformatted in ExFAT, did a clean core install, same deal CRC32s for the mentioned ROMs, for anyone interested
> 

First. I have The exact same Metroid-Zero-Mission ROM (5c61a844) and I have 0 problems... I can only guess the SD is a problem. That being said, I do have problems with Sleep Mode / Save States sometimes that don't load ...but various other people have reported this too (see Reddit).

### Reformat to FAT and use https://github.com/mattpannella/pocket-updater-utility/releases to properly setup the DIRs...

**ExFAT is notoriously unstable**. (Which is why, as a popular example, the Nintendo Switch SD back in the day of novelty, required a FAT formatted SD, not ExFAT).
You only need ExFat if you use ROMs (files) that are over 4GB in size (AFAIK). Since ROMs for this product are usually not over 32mb (unless I have overlooked something) FAT is fine.

**FAT format is the way to go for SD on Analogue Pocket in my opinion.**

_Stating the obvious:_ Did you try a second SD-Card (newly formatted to FAT)? As I have mentioned a) I am using the same ROM = 0 problems, b) you can see how non-often this core is updated, I doubt you won't receive a fix down the line any time soon (if ever), the next update of this core will drop when Analogue *finally* releases FPGA-Core-Display-Modes (which will be the main focus)

-- This is my SD Card fyi (it works flawlessly since 2022 for FPGA ...albeit not Save States / Sleep mode as indicated):

<img width="331" alt="Screenshot 2023-09-15 at 08 29 02" src="https://github.com/spiritualized1997/openFPGA-GBA/assets/6884752/6ad4a03b-9cf5-4dce-97fd-4a539afb65a3">

**Comment by azurehours** [comment]
Reformatted to FAT and still no luck, crashing/freezing in under a minute. I don't have a free second SD card on hand, but I'll order one and report back.
I did use Pocket Sync to install the cores, and double-checking with the updater utility you linked, the FS structure is the same.

**Comment by execution101** [comment]
I wish FAT32 would be the solution. I tried it just now same issue with it crashing in Doom 2. I have a friend that has an Analogue Pocket as well he got his in one of the first shipments that were sent he was the motivator for me to get one after trying his out. Anyways I recorded a video of both of our pockets using my sd card showing that his would work just fine while mine would not work. Here's a link to the video https://www.youtube.com/watch?v=m2cUZZxfq-I

I'm thinking my unit is defective or these newer pockets have some kind of hardware revision.

**Comment by azurehours** [comment]
Tried a new microSD, a SanDisk Extreme 32GB, same problem. Something I've noticed is that some games will freeze and the audio continues as normal, while others will freeze and the audio starts glitching.

**Comment by anamegoeshere23** [comment]
I don't think it's an sd card issue, I tried a couple cards and a number of configurations, all the same result.  I very much would like if this can be fixed through firmware/software update.

**Comment by execution101** [comment]
Yeah I don't think its an SD card issue as well. Either the core needs some updating do to a hardware revision or some of the pockets being shipped out are defective in some way with the hardware. I've been reaching out to Analogue Customer Service and they are so slow at responding.

**Comment by MarcoPNS** [comment]
Still in contact with the support. Unfortunately, no result so far. Neither Core nor Catridge GBA is playable for me. The transparent version will be coming soon and I will test it directly.

**Comment by execution101** [comment]
https://www.reddit.com/r/AnaloguePocket/comments/16jhd6n/comment/k3vrekj/?utm_source=share&utm_medium=mweb3x&utm_name=mweb3xcss&utm_term=1&utm_content=share_button

So this guy had a suggestion of using the 1.0 release of the core. I just put the 1.0 release on and so far with Doom 2 and Castlevania I haven't experienced any crashes. I didn't add this to my post since I hadn't tried playing it yet but playing Kirby and the Amazing Mirror it would crash after a few minutes of play. On the 1.0 I haven't had crashes in that Kirby game as well.

Can you guys try and see if that works for you?

**Comment by MarcoPNS** [comment]
I can confirm that v1.0.0 does not crash. I tested it with Mario Kart. I don't know why. I tried to compare the two versions and maybe because v1.2.0 does not have "variants"? Or maybe some of the options are buggy? Idk.

**Comment by jakubito** [comment]
I have the same issue with version 1.2.0, yellow Pocket. Final Fantasy 6 crashes right after starting a new game. Can confirm downgrading to 1.0.0 helped

**Comment by Computerlabrat** [comment]
It also has an issue with the Sonic SatAM fan game Sonic Epoch Advance. After you delete the existing save files and play, it seems it won't move on to level 2.
